$OpenBSD: patch-lib_common_layout_c,v 1.1.1.1 2001/07/13 17:50:45 pvalchev Exp $
The patch fixes:
1) <HR> placement
2) formatting inside <PRE></PRE> (not really nice HTML, but any browser
   supports it)
3) "Can't convert XmHTML_DEFAULT_GAMMA" runtime warning

--- lib/common/layout.c.orig	Thu Jan 28 01:10:17 1999
+++ lib/common/layout.c	Sat Jun 30 23:39:05 2001
@@ -2377,14 +2377,16 @@ SetRule(XmHTMLWidget html, PositionBox *
 	* Linefeeds in rules are divided accross the rule: half above and
 	* half below.
 	*****/
-	if(data->linefeed)
+	if(data->linefeed) {
+		box->y += data->linefeed;
 		y_offset = data->linefeed/2;
-	else
+	} else {
 		y_offset = data->font->height/2; 
+	}
 
 	data->y = box->y + y_offset;
-
-	/* take height of rule into account as well */
+	
+        /* take height of rule into account as well */
 	y_offset += data->height/2;
 
 	/* full height of the box */
